source/opt/local_ssa_elim_pass.cpp

```diff
@@ -0,0 +1,759 @@
+}
+
+void LocalSSAElimPass::SSABlockInitSinglePred(ir::BasicBlock* block_ptr) {
+  // Copy map entry from single predecessor
+  const uint32_t label = block_ptr->id();
+  const uint32_t predLabel = label2preds_[label].front();
+  assert(visitedBlocks_.find(predLabel) != visitedBlocks_.end());
+  label2ssa_map_[label] = label2ssa_map_[predLabel];
+}
+
+bool LocalSSAElimPass::IsLiveAfter(uint32_t var_id, uint32_t label) const {
+  // For now, return very conservative result: true. This will result in
+  // correct, but possibly usused, phi code to be generated. A subsequent
+  // DCE pass should eliminate this code.
+  // TODO(greg-lunarg): Return more accurate information
+  (void) var_id;
+  (void) label;
+  return true;
+}
+
+uint32_t LocalSSAElimPass::Type2Undef(uint32_t type_id) {
+  const auto uitr = type2undefs_.find(type_id);
+  if (uitr != type2undefs_.end())
+    return uitr->second;
+  const uint32_t undefId = TakeNextId();
+  std::unique_ptr<ir::Instruction> undef_inst(
+    new ir::Instruction(SpvOpUndef, type_id, undefId, {}));
+  def_use_mgr_->AnalyzeInstDefUse(&*undef_inst);
+  module_->AddGlobalValue(std::move(undef_inst));
+  type2undefs_[type_id] = undefId;
+  return undefId;
+}
+
+uint32_t LocalSSAElimPass::GetPointeeTypeId(
+    const ir::Instruction* ptrInst) const {
+  const uint32_t ptrTypeId = ptrInst->type_id();
+  const ir::Instruction* ptrTypeInst = def_use_mgr_->GetDef(ptrTypeId);
+  return ptrTypeInst->GetSingleWordInOperand(kTypePointerTypeIdInIdx);
+}
+
+void LocalSSAElimPass::SSABlockInitLoopHeader(
+    std::list<ir::BasicBlock*>::iterator block_itr) {
+  const uint32_t label = (*block_itr)->id();
+  // Determine backedge label.
+  uint32_t backLabel = 0;
+  for (uint32_t predLabel : label2preds_[label])
+    if (visitedBlocks_.find(predLabel) == visitedBlocks_.end()) {
+      assert(backLabel == 0);
+      backLabel = predLabel;
+      break;
+    }
+  assert(backLabel != 0);
+  // Determine merge block.
+  auto mergeInst = (*block_itr)->begin();
+  uint32_t mergeLabel = mergeInst->GetSingleWordInOperand(
+      kLoopMergeMergeBlockIdInIdx);
+  // Collect all live variables and a default value for each across all
+  // non-backedge predecesors. Must be ordered map because phis are
+  // generated based on order and test results will otherwise vary across
+  // platforms.
+  std::map<uint32_t, uint32_t> liveVars;
+  for (uint32_t predLabel : label2preds_[label]) {
+    for (auto var_val : label2ssa_map_[predLabel]) {
+      uint32_t varId = var_val.first;
+      liveVars[varId] = var_val.second;
+    }
+  }
+  // Add all stored variables in loop. Set their default value id to zero.
+  for (auto bi = block_itr; (*bi)->id() != mergeLabel; ++bi) {
+    ir::BasicBlock* bp = *bi;
+    for (auto ii = bp->begin(); ii != bp->end(); ++ii) {
+      if (ii->opcode() != SpvOpStore)
+        continue;
+      uint32_t varId;
+      (void) GetPtr(&*ii, &varId);
+      if (!IsTargetVar(varId))
+        break;
```

Comment:
Yeah, I am just trying to keep indentation down.

Anyway, good catch! This was a copy and paste error. I had copied out of a case statement where break *is* the thing you want to do.